tensorflow/python/kernel_tests/nn_ops/depthwise_conv_op_base.py

```diff
@@ -954,25 +954,26 @@ def testDepthwiseConv2DFilterGradExplicit(self):
 
   def _CompareBackpropInput(self, input_sizes, filter_sizes, output_sizes,
                             stride, padding, dtype):
-    x1 = np.random.rand(*filter_sizes).astype(dtype)
-    x2 = np.random.rand(*output_sizes).astype(dtype)
+    x1 = np.random.rand(*filter_sizes)
+    x2 = np.random.rand(*output_sizes)
     if isinstance(padding, list):
       padding = [(0, 0)] + padding + [(0, 0)]
 
-    def _GetVal(use_gpu):
+    def _GetVal(use_gpu, dtype):
       with self.cached_session(use_gpu=use_gpu):
         t0 = constant_op.constant(input_sizes, shape=[len(input_sizes)])
-        t1 = constant_op.constant(x1, shape=filter_sizes)
-        t2 = constant_op.constant(x2, shape=output_sizes)
+        t1 = constant_op.constant(x1, shape=filter_sizes, dtype=dtype)
+        t2 = constant_op.constant(x2, shape=output_sizes, dtype=dtype)
         backprop = nn_ops.depthwise_conv2d_native_backprop_input(
             t0, t1, t2, strides=[1, stride, stride, 1], padding=padding)
         ret = self.evaluate(backprop)
         self.assertShapeEqual(ret, backprop)
         return ret
 
-    gpu_value = _GetVal(use_gpu=True)
-    cpu_value = _GetVal(use_gpu=False)
-    self.assertAllClose(cpu_value, gpu_value, rtol=1e-4, atol=1e-4)
+    rtol, atol = (1e-2, 1e-2) if dtype == 'bfloat16' else (1e-4, 1e-4)
+    gpu_value = _GetVal(use_gpu=True, dtype=dtype)
+    cpu_value = _GetVal(use_gpu=False, dtype=dtype)
+    self.assertAllClose(cpu_value, gpu_value, rtol=rtol, atol=atol)
 
```

Comment:
Can you also test bf16 in the functions that don't compare cpu and gpu? Namely:
* `testDepthwiseConv2DCudnn`
* `testDepthwiseConv2DExplicit`
* `testDepthwiseConv2DInputGradCudnn`
* `testDepthwiseConv2DInputGradExplicit`
* `testDepthwiseConv2DFilterGradCudnn`
* `testDepthwiseConv2DFilterGradExplicit`

This would be useful in case CPU and GPU both return the same incorrect results.

Reply:
It seems all these tests are `run_v1_only` and are skipped when I tried to run them. Do you know any methods to run them in TF2 container?

Reply:
I think setting the environment variable `TF2_BEHAVIOR=0` might allow them to run. Alternatively, passing `--config=v1` might work, but this might trigger a rebuild.